Adding a threshold setting to FGFXLargeScalePerceptualObscuranceIrradiance.fx

snyviper opened the pull request:
FGFXLargeScalePerceptualObscuranceIrradiance.fx originally detects anything brighter than 0.5 as Irradiance and anything darker as Occlusion. Some games look too dark to have the _Occlusion Intensity_ on anything higher than 0.

**I added an _Occlusion/Irradiance Threshold_ setting so that you can change the line that separates Occlusion and Irradiance to any value ranging from 0.004 to 0.996.**

I did notice you have an _Occlusion-Irradiance Neutral Point_ hidden setting, but it only seems to increase/decrease the brightness. It is different from my solution, which again, tweaks the middle line between Occlusion and Irradiance.

I've been using this tweaked version of your shader in many games, and I figured it would be more convenient if this was accepted in the main file, plus other people could have access to it too.

As an extra suggestion, I would also love to have the default value of _Effect Saturation_ set to _0.1_ since it's the minimum I've used ever since I found out how beautiful games can look with it.

snyviper committed:
Update FGFXLargeScalePerceptualObscuranceIrradiance.fx

Adding a threshold setting to FGFXLargeScalePerceptualObscuranceIrradiance.fx

snyviper committed:
Merge pull request #1 from snyviper/FGFXLSPOI-threshold

Add threshold to FGFXLargeScalePerceptualObscuranceIrradiance.fx

AlexTuduran commented:
Please change change the name of "ScaleOcclusionAndIrradiance2" to something more meaningful (ex. "ThresholdedScaleOcclusionAndIrradiance"). As for the Effect Saturation, set its default to 0.1 if you like it more.

AlexTuduran commented:
Also please use spaces before and after '/' in "Occlusion/Irradiance Threshold".

snyviper committed:
Update FGFXLargeScalePerceptualObscuranceIrradiance.fx

Updated parameter naming and set default Effect Saturation as 0.1

snyviper committed:
Merge pull request #2 from snyviper/FGFXLSPOI-threshold

Update FGFXLargeScalePerceptualObscuranceIrradiance.fx

snyviper committed:
Update FGFXLargeScalePerceptualObscuranceIrradiance.fx

Swapping ThresholdedScaleOcclusionAndIrradiance with ScaleOcclusionAndIrradiance

snyviper committed:
Merge pull request #3 from snyviper/FGFXLSPOI-threshold

Update FGFXLargeScalePerceptualObscuranceIrradiance.fx

snyviper commented:
Changes made! 😄

snyviper committed:
Update FGFXLargeScalePerceptualObscuranceIrradiance.fx

Improving readability of ThresholdedScaleOcclusionAndIrradiance formulas.

snyviper committed:
Merge pull request #4 from snyviper/FGFXLSPOI-threshold

Update FGFXLargeScalePerceptualObscuranceIrradiance.fx

snyviper committed:
Update FGFXLargeScalePerceptualObscuranceIrradiance.fx

Standardizing identation

snyviper committed:
Merge pull request #5 from snyviper/FGFXLSPOI-threshold

Update FGFXLargeScalePerceptualObscuranceIrradiance.fx

snyviper commented:
The extra changes were just to improve the readability of the thresholded formulas and to standardize the indentation based on the rest of the code 🙂

snyviper commented:
@AlexTuduran there's nothing else I want to add or change, so unless you want me to do something, this should be ready to go

AlexTuduran commented:
That's great mate. Thanks for the contribution!
